src/services/order.ts

```diff
@@ -22,27 +29,45 @@ export class OrderService {
     order: OrderModel,
     orderRequest: Omit<OrderRequest, 'timestamp'>
   ) {
-    for (const productId in orderRequest.items) {
-      const product = await this.productModel.findOneById(productId);
+    for (const orderItem of orderRequest.items) {
+      const product = await this.productModel.findOneById(orderItem.productId);
+
+      const additives = [];
+
+      for (const additiveId in orderItem.additives) {
+        const additive = await this.additiveModel.findOneById(additiveId);
+        if (additive) {
+          additives.push({
+            id: additiveId,
+            name: additive.name,
+            price: additive.price,
+            count: orderItem.additives[additiveId]
+          });
+        } else {
+          throw new Error(`Additive with id: ${additiveId} not found`);
+        }
+      }
 
       if (product) {
         order.items.push({
-          id: productId,
+          id: orderItem.productId,
           name: product.name,
           price: product.price,
-          count: orderRequest.items[productId]
+          additives: additives
         });
       } else {
-        throw new Error(`Product with id: ${productId} not found`);
+        throw new Error(`Product with id: ${orderItem.productId} not found`);
```

Comment:
The same as https://github.com/shamanms/vash-lavash-api/pull/24/files#r1009884223